tests/tcp.c

```diff
@@ -178,6 +178,28 @@ int main ()
     test_close (sc);
     test_close (s1);
 
+    /*  Test NN_RCVMAXSIZE */
+    sb = test_socket (AF_SP, NN_PAIR);
+    test_bind (sb, SOCKET_ADDRESS);
+    s1 = test_socket (AF_SP, NN_PAIR);
+    test_connect (s1, SOCKET_ADDRESS);
+    opt = 4;
+    nn_setsockopt(sb, NN_SOL_SOCKET, NN_RCVMAXSIZE, &opt, sizeof (opt));
+    nn_sleep (100);
+    test_send (s1, "ABC");
+    test_recv (sb, "ABC");
+    test_send (s1, "0123456789012345678901234567890123456789");
+    void * dummy_buf;
+    int s;
+    s = nn_recv(sb, dummy_buf, NN_MSG, NN_DONTWAIT);
+    if (s != -1 || errno != EAGAIN) {
```

Comment:
Rather, here is an example of a conventional assert for nanomsg tests: https://github.com/nanomsg/nanomsg/blob/master/tests/tcp.c#L78-L80